internal/controller/container_reconciler.go

```diff
@@ -58,11 +64,96 @@ func (r *ContainerImageReconciler) ReconcileContainerImage(ctx context.Context,
 		return result, err
 	}
 
-	// The Job that will build the container image.
-	buildJob, err := r.buildJob(ctx, obj)
-	if err != nil {
-		log.Error(err, "unable to construct image-builder Job")
-		// No use in retrying...
+	var buildJob *batchv1.Job
+
+	if specUpload := obj.GetImage().Upload; specUpload != nil && specUpload.Md5Checksum != "" {
+		statusMd5, statusUploadURL := obj.GetStatusUpload().Md5Checksum, obj.GetStatusUpload().UploadURL
+
+		// an upload object md5 has been declared and doesn't match the current spec
+		// generate a signed URL
+		if specUpload.Md5Checksum != statusMd5 {
+			url, err := r.callSignedUrlGenerator(obj, r.CloudManagerGrpcClient)
+			if err != nil {
+				return result{}, fmt.Errorf("generating upload url: %w", err)
+			}
+
+			obj.SetStatusUpload(ssv1.UploadStatus{
+				UploadURL:   url,
+				Md5Checksum: specUpload.Md5Checksum,
+			})
+			meta.SetStatusCondition(obj.GetConditions(), metav1.Condition{
+				Type:               apiv1.ConditionUploaded,
+				Status:             metav1.ConditionFalse,
+				Reason:             apiv1.ReasonUploadIncomplete,
+				ObservedGeneration: obj.GetGeneration(),
+				Message:            fmt.Sprintf("Waiting for object upload to complete: %v", obj.GetName()),
+			})
+			if err := r.Client.Status().Update(ctx, obj); err != nil {
+				return result{}, fmt.Errorf("updating status: %w", err)
+			}
+			return result{}, nil
+		}
+
+		// if the upload URL has expired, clear it from the status leaving the md5 checksum
+		if statusUploadURL != "" {
+			expirationTime, err := r.getExpirationTime(statusUploadURL)
+			if err != nil {
+				return result{}, fmt.Errorf("getting URL expiration time: %w", err)
+			}
+
+			if time.Now().After(expirationTime) {
+				log.Info("The signed URL has expired. Clearing .Status.UploadURL")
+				// TODO(bjb): why doesn't this work?
+				obj.SetStatusUpload(ssv1.UploadStatus{
+					UploadURL:   "",
+					Md5Checksum: statusMd5,
+				})
+				return result{}, nil
```

Comment:
@nstogner - this doesn't seem to work. I see the log message emitted but the status isn't cleared.

Reply:
Missing call to `.Status().Update`